modal/_container_entrypoint.py

```diff
@@ -309,18 +457,18 @@ def _cancel_input_signal_handler(signum, stackframe):
                 # run this sync code in the main thread, blocking the "userland" event loop
                 # this lets us cancel it using a signal handler that raises an exception
                 try:
-                    run_input_sync(input_id, function_call_id, args, kwargs)
+                    run_input_sync(finalized_function, input_id, function_call_id, args, kwargs)
                 finally:
                     signal.signal(signal.SIGUSR1, usr1_handler)  # reset signal handler
 
 
-def import_function(
+def import_single_function_service(
```

Comment:
What's the "single function" terminology mean? Is that the same thing as a "function that isn't a class"? I guess not though as I would expect the return type to be an `ImportedFunction` in that case.

Reply:
Yeah it's a funciton that isn't a class, i.e. all old function definitions. The runtime returned object type is `ImportedFunction` but it's type annotated as a `Service` (its abstract baseclass) to indicate that anything that's not part of the Service interface shouldn't be used by the calling code.